### DISCUSSION TITLE
Digital Asset Standard not working

### DISCUSSION THREAD

**Discussion by memxor** [aptos-labs/aptos-developer-discussions#420] [Questions] [answered]
### Discord user ID

memxor

### Describe your question in detail.

I'm trying to create an in-game characters minting system, where multiple different types of character NFTs go into the Character Collection. I want these NFTs to be tradeable in the marketplace, so I created a Collection in the `init_module`.

```cs
        let collection = string::utf8(b"CC Characters");
        collection::create_unlimited_collection(
            admin,
            string::utf8(b"The brave dungeon raiders."),
            collection,
            option::none(),
            string::utf8(b"https://akrd.net/LniukunI-rP_qa_rn7SD4hIa6XB29Db6AjssvH5LFJE")
        );

        move_to(admin, OnChainConfig {
            collection,
        });
```

I also have an Array of all the pre-determined character values in the `init_module`. Something like this 
```cs
let characters: vector<Character> = 
        vector<Character>[
            Character
            {
                name: string::utf8(b"Rad Wraith"),
                description: string::utf8(b"Rad Wraith, a post-apocalyptic survivor with a cowboy hat, masters the shotgun for brutal close-quarters combat."),
                movingSpeed: 8,
                rollSpeed: 9,
                rollDistance: 8,
                maxHealth: 100,
                price: 0,
                uri: string::utf8(b"https://akrd.net/NiqFHcPSRLvY0Opv2TNGQE_M-K8HOVjPCwSyblXbfu8"),

            },
            .
            .
            .
        ];
        
        move_to(admin, Characters { characters });
```

Which I then access from the minting function. Here is the minting function - 
```cs
public entry fun mint_character(creator: &signer, type: u64) acquires Characters, OnChainConfig
    {
        let characters = borrow_global<Characters>(@publisher).characters;
        let character = vector::borrow(&characters, type);

        let on_chain_config = borrow_global<OnChainConfig>(@publisher);
        let constructor_ref = token::create_named_token(
            creator,
            on_chain_config.collection,
            character.description,
            character.name,
            option::none(),
            character.uri
        );
        let token_signer = object::generate_signer(&constructor_ref);

        move_to(&token_signer, *character);

        if(character.price > 0)
        {
            cryptara_conquest_token::burn(signer::address_of(creator), character.price);
        }
    }
```

When I publish and run the mint function, I get the following error - 
![image](https://github.com/user-attachments/assets/029e912d-01ce-499f-9df0-6cfed0a93226)

What is causing this issue?

### What error, if any, are you getting?

Simulation Error - 
An object doesn't exist in this address

I tried changing the `@publisher` to `@resourse_publisher` in `mint_character`
![image](https://github.com/user-attachments/assets/12585104-259f-45da-a5dd-139b494e42a3)
![image](https://github.com/user-attachments/assets/40863e2f-5e36-4941-9a05-cd4f311b4f4f)

Doing this gives me a new error: Simulation Failed - Generic error

**Comment by 0xbrw** [comment]
The error message is describing the object you are trying to retrieve does not exist, this may have to do with a mistake during collection creation.

To fully debug you error would be difficult to do without access to your entire codebase, I would attempt to reference this example during collection creation (cc @0xaptosj): https://github.com/aptos-labs/move-by-examples/blob/main/nft-launchpad/aptos/move/sources/launchpad.move#L161-L282

And reference this code during minting: https://github.com/aptos-labs/move-by-examples/blob/main/nft-launchpad/aptos/move/sources/launchpad.move#L286-L313

**Reply by 0x-j** [reply]
@memxor Yes it should be easier to start your contract from the nft minting template Brian linked instead of writing from scratch

if you are looking for a simpler version where the contract only controls a single collection, take a look at https://github.com/aptos-labs/aptogotchi

**Comment by memxor** [comment]
I've updated the minting function as provided by @bwdotcom.. This is my latest minting code - 

```cs
      public entry fun mint_character(creator: &signer, type: u64) acquires Characters, OnChainConfig
    {
        let resource_publisher = cryptara_conquest_publisher_signer::get_signer();
        let characters = borrow_global<Characters>(signer::address_of(&resource_publisher)).characters;
        let character = vector::borrow(&characters, type);

        let on_chain_config = borrow_global<OnChainConfig>(signer::address_of(&resource_publisher));
        let constructor_ref = token::create(
            &resource_publisher,
            on_chain_config.collection,
            character.description,
            character.name,
            option::none(),
            character.uri
        );

        let nft = object::object_from_constructor_ref<Token>(&constructor_ref);
        object::transfer(&resource_publisher, nft, signer::address_of(creator));

        if(character.price > 0)
        {
            cryptara_conquest_token::burn(signer::address_of(creator), character.price);
        }
    }
```

It compiles fine. But it's not running, when I try to run it in the explorer, it gives me Generic Error
![image](https://github.com/user-attachments/assets/d863c64a-7b85-4467-a442-3c3a16c10b15)

Error in transaction - `Execution failed in 0x465b1cf749e60c26f1a21ab6d3b33422785b6f2c42a3e226e24690b9a587301::cryptara_conquest_characters::mint_character at code offset 1`

here is the explorer if anyone wanna play with it and check - https://explorer.aptoslabs.com/account/0x0465b1cf749e60c26f1a21ab6d3b33422785b6f2c42a3e226e24690b9a587301/modules/run/cryptara_conquest_characters/mint_character?network=testnet

**Reply by memxor** [reply, marked as answer]
This issue is fixed, it was due to mismatching accounts
